requests/models.py

```diff
@@ -439,9 +439,7 @@ def prepare_body(self, data, files, json=None):
             if files:
                 raise NotImplementedError('Streamed bodies and files are mutually exclusive.')
 
-            if length:
-                self.headers['Content-Length'] = builtin_str(length)
-            else:
+            if not length:
```

Comment:
We very deliberately decided, several months ago, to handle things that appear to have zero length by chunk-encoding them. This protects against the possibility that we computed a zero length for something that was in fact length undefined.